src/charm.py

```diff
@@ -174,6 +186,106 @@ def _on_register_user_action(self, event: ActionEvent) -> None:
         results = {"register-user": True, "user-password": user.password}
         event.set_results(results)
 
+    def _create_admin_user(self) -> typing.Optional[User]:
+        """Create admin user.
+
+        Returns:
+            Admin user with token to be used in Synapse API requests or None if fails.
+        """
+        container = self.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+        if not container.can_connect():
+            logger.error("Failed to create admin user: waiting for pebble")
```

Comment:
Should it be "Failed to create admin user: failed to connect to the container" ?

Reply:
Good catch. To not lose your approval, I'll fix this in my other open PR [here](https://github.com/canonical/synapse-operator/pull/77).

Reply:
Since I had to add more unit test,  I fixed this here, thanks.